strategies/key.test.js

```diff
@@ -1,64 +1,131 @@
 'use strict';
+var mockUser, OLD_ACCESS_KEY;
+
+jest.mock('../utils', () => {
+  return {
+    ...jest.requireActual('../utils'),
+    fetchUserViaAPIKey: jest.fn().mockResolvedValue(mockUser)
+  };
+});
 
 const _startCase = require('lodash/startCase'),
   _includes = require('lodash/includes'),
-  passport = require('passport'),
+  bluebird = require('bluebird'),
   filename = __filename.split('/').pop().split('.').shift(),
-  lib = require(`./${filename}`);
+  utils = require('../utils');
 
 describe(_startCase(filename), function () {
+  beforeAll(() => {
+    OLD_ACCESS_KEY = process.env.CLAY_ACCESS_KEY;
+  });
+
+  beforeEach(() => {
+    mockUser = null;
+    process.env.CLAY_ACCESS_KEY = '123';
+    jest.resetModules();
+    utils.fetchUserViaAPIKey = jest.fn().mockResolvedValue(mockUser);
+  });
+
+  afterAll(() => {
+    process.env.CLAY_ACCESS_KEY = OLD_ACCESS_KEY;
```

Comment:
not sure i understand what OLD_ACCESS_KEY is doing here, is it insufficient to just set `process.env.CLAY_ACCESES_KEY` as in L24 above?

Reply:
This is really just a pattern to preserve the original values of the environment before the suite was executed.

I think ideally we'd mock out the whole `process.env` for tests for reproducibility, but chose to stick close to what we had for this PR.